packages/likert/src/index.js

```diff
@@ -1,58 +1,97 @@
+import Main from './main';
 import React from 'react';
 import ReactDOM from 'react-dom';
-import Main from './main';
-import { SessionChangedEvent } from '@pie-framework/pie-player-events';
+import debounce from 'lodash/debounce';
+import debug from 'debug';
 import { renderMath } from '@pie-lib/math-rendering';
+import { updateSessionValue } from './session-updater';
+
+const log = debug('pie-ui:likert');
 
-export { Main as Component };
+export const isComplete = session => !!(session && session.value && session.value.length);
 
 export default class Likert extends HTMLElement {
   constructor() {
     super();
+    this._model = null;
+    this._session = null;
+
+    this._rerender = debounce(
+      () => {
+        if (this._model && this._session) {
+          const element = React.createElement(Main, {
+            model: this._model,
+            session: this._session,
+            onChoiceChanged: this._onChange.bind(this)
+          });
+          ReactDOM.render(element, this, () => {
+            log('render complete - render math');
+            renderMath(this);
+          });
+        } else {
+          log('skip');
+        }
+      },
+      50,
+      { leading: false, trailing: true }
+    );
+
+    this._dispatchResponseChanged = debounce(() => {
+      const event = new CustomEvent('session-changed', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          complete: isComplete(this._session),
+          component: this.tagName.toLowerCase()
+        }
+      });
+
+      this.dispatchEvent(event);
+    });
+
+    this._dispatchModelSet = debounce(
+      () => {
+        this.dispatchEvent(
+          new CustomEvent('model-set', {
```

Comment:
use the events from this package: https://github.com/pie-framework/pie-player-events

Reply:
solved